src/stavenote.js

```diff
@@ -725,6 +739,7 @@ export class StaveNote extends StemmableNote {
   setKeyLine(index, line) {
     this.keyProps[index].line = line;
     this.note_heads[index].setLine(line);
+    this.reset();
```

Comment:
This change is important. `StaveNote#reset()` rebuilds the `NoteHeads`, which clearly must be done after changing the line of a key. While this is a bug fix, it affects some of the test output in a negative way.